therapy/etl/hemonc.py

```diff
@@ -0,0 +1,233 @@
+        """Download HemOnc.org source data.
+
+        Raises download exception for now -- HTTP authorization may be
+        possible?
+        """
+        raise DownloadException("No download for HemOnc data available -- "
+                                "must place manually in data/ directory.")
+
+    def _extract_data(self):
+        """Get source files from data directory."""
+        self._src_data_dir.mkdir(exist_ok=True, parents=True)
+        self._src_files = []
+        for item_type in ('concepts', 'rels', 'synonyms'):
+            src_file_prefix = f'hemonc_{item_type}_'
+            dir_files = [f for f in self._src_data_dir.iterdir()
+                         if f.name.startswith(src_file_prefix)]
+            if len(dir_files) == 0:
+                self._download_data()
+                dir_files = [f for f in self._src_data_dir.iterdir()
+                             if f.name.startswith(src_file_prefix)]
+            self._src_files.append(sorted(dir_files, reverse=True)[0])
+        self._version = self._src_files[0].stem.split('_', 2)[-1]
+
+    def _load_meta(self):
+        """Add HemOnc metadata."""
+        meta = {
+            'data_license': 'CC BY 4.0',
+            'data_license_url': 'https://creativecommons.org/licenses/by/4.0/legalcode',  # noqa: E501
+            'version': self._version,
+            'data_url': 'https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/9CY9C6',  # noqa: E501
+            'rdp_url': None,
+            'data_license_attributes': {
+                'non_commercial': False,
+                'share_alike': False,
+                'attribution': True,
+            },
+        }
+        assert SourceMeta(**meta)
+        meta['src_name'] = SourceName.HEMONC.value
+        self.database.metadata.put_item(Item=meta)
+
+    def _get_concepts(self) -> Tuple[Dict, Dict, Dict]:
+        """Get therapy, brand name, and disease concepts from concepts file.
+        :return: Tuple of dicts mapping ID to object for each type of concept
+        """
+        therapies = {}  # hemonc id -> record
+        brand_names = {}  # hemonc id -> brand name
+        conditions = {}  # hemonc id -> condition name
+
+        concepts_file = open(self._src_files[0], 'r')
+        concepts_reader = csv.reader(concepts_file)
+        next(concepts_reader)  # skip header
+        for row in concepts_reader:
+            if row[6]:
+                continue  # skip if deprecated/invalid
+
+            row_type = row[2]
+            if row_type == 'Component':
+                concept_id = f'{NamespacePrefix.HEMONC.value}:{row[3]}'
+                therapies[row[3]] = {
+                    'concept_id': concept_id,
+                    'label': row[0],
+                    'trade_names': [],
+                    'aliases': [],
+                    'other_identifiers': [],
+                }
+            elif row_type == 'Brand Name':
+                brand_names[row[3]] = row[0]
```

Comment:
I haven't checked this, but you probably have. Is it possible that a concept could have multiple unique brand names?

Reply:
yep! here, `row[3]` is the ID for the brand name (string in `row[0]`). The relations CSV then ties this ID back to the ID for the concept.